### DISCUSSION TITLE
Kalker integrated into Helix text editor

### DISCUSSION THREAD

**Discussion by timokoesters** [PaddiM8/kalk#148] [Show and tell]
Hey, Kalker is very fast and easy to use, so I added it as a keybind to my text editor Helix. It will append the Kalker result when I press Ctrl+k, here's a video:

https://github.com/PaddiM8/kalker/assets/25297359/f8b27b08-1bdd-4549-99ee-d3eee99a3592

Thanks for the tool!

The code got pretty messy because Helix doesn't support scripting very well yet, but here is the code anyway if someone wants to build something similar

```
"C-k" = [
  "move_char_right", # Select whole set of equations
  "goto_prev_paragraph",
  "goto_next_paragraph",
  "trim_selections", # Remove empty lines
  "extend_to_line_bounds", # Include \n
  # Trim after last equal sign, then save input into x and give it to both echo and kalker
  ":pipe sd ' ?=.*\\n?\\z' '' | { x=$(cat); echo \"$x\" = \"$(kalker \"$x\" 2>&1)\"; }", 
  ":append-output echo -n =", # Move cursor back to last equal sign
  "search_selection",
  "delete_selection",
  "search_prev",
]
```

**Comment by PaddiM8** [comment]
That is _super_ cool! Thank you for sharing. Wow
